contents/ui/main.qml

```diff
@@ -53,4 +53,12 @@ Item {
         id: layoutOsdLoader
     }
 
+    Timer {
+        id: timer
+        repeat: false
+        interval: 1
```

Comment:
Which unit is this?

Reply:
Should be in milliseconds:
https://doc.qt.io/qt-5/qml-qtqml-timer.html#interval-prop

Reply:
Should be okay then. I thought it might be seconds.

Reply:
Typical solution in JS when using timer as a way to force async is to use 0 as a timeout value. However there's no information on this in the documentation linked by @laloch and I couldn't find it anywhere else neither.

Reply:
Well, we _do_ need to have a teensy bit of time elapse, because we have to let KWin get done with this pixmap before we tell it to resize again (I'm pretty sure that doesn't mean it has to be done with the _frame_, just one particular check).

We don't actually care about it being async. In fact I'd love it if it weren't.

Reply:
I used 1, because the behavior is not well documented and I needed to make sure, that the timer wan't fire from within the `start()` method immediately.

> We don't actually care about it being async. In fact I'd love it if it weren't.

True, but that would mean a lot of work in KWin and the change would probably not pass the review. I would have to change the API related to (at least) `geometryShapeChanged` and `geometryChanged`. The current implementation is such, that anyone who's inheriting from `AbstractClient` needs to take care to reflect any eventual geometry changes done in handler of any signal by himself. That's simply not robust enough to account on.

Reply:
The whole thing is still work around for bug https://bugs.kde.org/show_bug.cgi?id=341195, right?
Can we somehow bump that bug?